pkg/gui/controllers/basic_commits_controller.go

```diff
@@ -280,6 +281,14 @@ func (self *BasicCommitsController) createResetMenu(commit *models.Commit) error
 }
 
 func (self *BasicCommitsController) checkout(commit *models.Commit) error {
+	commitBranches := lo.Filter(self.c.Model().Branches, func(branch *models.Branch, _ int) bool {
+		return commit.Hash == branch.CommitHash && branch.Name != self.c.Model().CheckedOutBranch
+	})
```

Comment:
Forgot to mention this last time: I'd move this code to the `CreateCheckoutMenu` method as well, there's no reason to do this here and pass it in. The general rule of thumb is to have as little logic as possible in controllers.

Reply:
Makes sense to me. Thanks for the rule of thumb, that'll help me in the future to get the abstraction right.

Reply:
Refactored.